addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2331,6 +2346,43 @@ export class OdooEditor extends EventTarget {
                 }
             }
         } else if (node.nodeType !== Node.TEXT_NODE) {
+            if (node.nodeName === 'TD' && (node.hasAttribute('style') || node.hasAttribute('bgcolor'))) {
+                // TD tags do not support style in v15, move style to child or new SPAN.
+                const keepChild = (node.childNodes.length === 1 && node.childNodes[0].nodeName === 'SPAN') 
+                const styleElem = keepChild ? node.childNodes[0] : this.document.createElement('SPAN');
+                if (node.hasAttribute('bgcolor') && !styleElem.style.cssText.includes('background-color:')) {
+                    // bgcolor attribute is deprecated, replaced with background-color.
+                    styleElem.setAttribute('style', 
+                        `${styleElem.getAttribute('style') || ''};background-color:${node.getAttribute('bgcolor')}`);
+                }
+                styleElem.setAttribute('style', `${styleElem.getAttribute('style') || ''};${node.getAttribute('style') || ''}`);
```

Comment:
I'm a little bit confused because for background color you specifically check that the span doesn't already have a background color, but here you're mercilessly overriding styles on the span with the ones from the td.